typing/typecore.ml

```diff
@@ -4783,10 +4783,13 @@ let type_clash_of_trace trace =
 (* Hint on type error on integer literals
    To avoid confusion, it is disabled on float literals
    and when the expected type is `int` *)
-let report_literal_type_constraint ppf expected_type const =
+let report_literal_type_constraint expected_type const =
   let hint str_val =
```

Comment:
I'm not so fond of the structure of the code here, with the nested `hint` and `hint_suffix` and the partial applications.
Could this be written instead as something like:

```ocaml
let report_literal_type_constraint expected_type const =
  let const_str = match const with
    | Const_int n -> Some (Int.to_string n)
    | ...
    | _ -> None
  in
  let suffix =
    if Path.same expected_type Predef.path_int32 then
      Some 'l'
    else ... else None
  in
  match const_str, suffix with
  | Some c, Some s -> [ Location.msg "..." c s ]
  | _ -> []
```

I think it would make the code more straightforward.

Reply:
I agree this is better.
I tried to apply the same logic to the next function (`report_numeric_operator_clash_hints`). I'll revert if this is not good or if it's not the scope of this PR.